### PR TITLE
Issue/233 welsh localisation

### PR DESCRIPTION
**Pull request by ijdickinson**
This is the first tranche of changes for #233. In this PR, we:

- extract English text from Rails templates, and move it to the `config/locale/en.yml` message catalogue
- add Vue localisation support
- arrange for webpack to read `config/locale/en.yml` so that the JavaScript code can also perform localisation using the same message catalogue
- extract text from Vue templates and code to the message catalogue
- make the headers for the landing page and app pages of UKHPI the same
- add an `English | Cymraeg` language switching affordance

### CODE REVIEW

File: app/views/about/index.html.haml [hunk 1]
```diff
@@ -0,0 +1 @@
+%h1 About UKHPI
```
Comment (joescottdave) [inline app/views/about/index.html.haml:1]: Placeholder?

File: app/javascript/components/compare-selection.vue [hunk 2]
```diff
@@ -28,9 +28,9 @@ export default {
     },
 
     prompt() {
-      const label = this.selectedLocation ? this.selectedLocation.labels.en : '';
+      const label = this.selectedLocation ? this.selectedLocation.labels[this.$locale] : '';
       return `Select an area to compare against ${label}`;
```
Comment (bogdanadrianmarc) [inline app/javascript/components/compare-selection.vue:32]: Was this message intentionally left out?

File: app/javascript/components/data-view-dates.vue [hunk 3]
```diff
@@ -6,7 +6,9 @@
         class='c-options-selection__button'
         title='change start or end date'
```
Comment (bogdanadrianmarc) [inline app/javascript/components/data-view-dates.vue:7]: Same here for the title

File: app/views/compare/show.html.haml [hunk 4]
```diff
@@ -43,7 +48,7 @@
               Add another location:
```
Comment (bogdanadrianmarc) [inline app/views/compare/show.html.haml:48]: Also here

File: app/views/compare/show.html.haml [hunk 5]
```diff
@@ -43,7 +48,7 @@
...
               %input{ type: 'text', name: 'location-term', value: @view_state.search_term }
               %button{ type: 'submit', value: 'search', name: 'form-action' }
-                Search
+                = t('action.search')
 
           - if @view_state.search_results
             %p Search results:
```
Comment (bogdanadrianmarc) [inline app/views/compare/show.html.haml:54]: And here

File: app/views/landing/index.html.haml [hunk 6]
```diff
@@ -15,24 +15,24 @@
...
       compared to the
       %span
         previous month,
       and
       %span.c-headline-figure.c-headline-figure__annual-change
-        = @landing_state.percentage_annual_change
+        = @view_state.percentage_annual_change
       compared to the
       %span
         previous year.
```
Comment (bogdanadrianmarc) [inline app/views/landing/index.html.haml:38]: Not sure why there aren't any translations in this file

File: app/views/about/index.html.haml (hunk 1 above)
Comment (ijdickinson) [reply app/views/about/index.html.haml:1]: Hmm, not sure what that is. Will investigate!

File: app/javascript/components/compare-selection.vue (hunk 2 above)
Comment (ijdickinson) [reply app/javascript/components/compare-selection.vue]: Not intentional. Have added a fix

File: app/javascript/components/data-view-dates.vue (hunk 3 above)
Comment (ijdickinson) [reply app/javascript/components/data-view-dates.vue]: Fix added

File: app/views/compare/show.html.haml (hunk 4 above)
Comment (ijdickinson) [reply app/views/compare/show.html.haml]: Fix added

File: app/views/compare/show.html.haml (hunk 5 above)
Comment (ijdickinson) [reply app/views/compare/show.html.haml]: Fix added

**Comment by ijdickinson** [comment]
Thanks @bogdanadrianmarc and @joescottdave. I've made some commits to fix the smaller omissions. Regarding the index page, I think we'll have to have the translator do that in one go. So I've added a couple of partials, one for :en and one for :cy, and some simple logic to select the right one. But the page will only show one `[cy]` in the heading

File: app/javascript/components/compare-selection.vue (hunk 2 above)
Comment (joescottdave) [reply app/javascript/components/compare-selection.vue]: Will the return statement also need translating?

**Review by joescottdave — APPROVED** [review]
All looks okay 🆗  noting that `/landing/index.html.haml` is missing translations for multiple lines of text, which is what I understand you refer to in your last comment.

**Review by bogdanadrianmarc — APPROVED** [review]
I must say that it is hard to review without going through everything again (doing so shows outdated changes), but I managed to do it commit by commit and everything looks alright 👍

**Comment by ijdickinson** [comment]
Thanks both. 

@joescottdave I think the return statement you highlighted has already been addressed in a later commit.

**Pull request merged**